vortex-array/src/arrays/varbin/mod.rs

```diff
@@ -213,30 +199,27 @@ impl VarBinArray {
 
         // Validate UTF-8 for Utf8 dtype
         if matches!(dtype, DType::Utf8(_)) {
-            let n_strings = offsets.len() - 1;
-            let sample_size = n_strings.min(100); // Check first 100 strings
-
-            for i in 0..sample_size {
-                if validity.is_null(i) {
-                    continue;
-                }
-
-                let start = offsets
-                    .scalar_at(i)
-                    .as_primitive()
-                    .as_::<usize>()
-                    .ok_or_else(|| vortex_err!("Offset must be convertible to usize"))?;
-                let end = offsets
-                    .scalar_at(i + 1)
-                    .as_primitive()
-                    .as_::<usize>()
-                    .ok_or_else(|| vortex_err!("Offset must be convertible to usize"))?;
-
-                let string_bytes = &bytes.as_ref()[start..end];
-                if std::str::from_utf8(string_bytes).is_err() {
-                    vortex_bail!("Invalid UTF-8 at index {}", i);
+            let primitive_offsets = offsets.to_primitive();
+            match_each_integer_ptype!(primitive_offsets.dtype().as_ptype(), |O| {
+                let offsets_slice = primitive_offsets.as_slice::<O>();
+                for (i, (start, end)) in offsets_slice
+                    .windows(2)
+                    .map(|o| (o[0].as_(), o[1].as_()))
+                    .enumerate()
+                {
+                    if validity.is_null(i) {
+                        continue;
+                    }
+
+                    let string_bytes = &bytes.as_ref()[start..end];
+                    simdutf8::basic::from_utf8(string_bytes).map_err(|_| {
```

Comment:
why do we not validate the whole array vs just part?

Does this make the slice operations non-constant